### PR TITLE
add did-frame-navigate event to WebContents, pass http response code to it and did-navigate

### PR DESCRIPTION
**Pull request by bughit**
This addresses two issues:

1. there is no did-navigate equivalent for frames
2. there is no event on WebContents that can notify of http level frame load failures, did-*-navigate is a good opportunity to pass the frame load http status code.

### CODE REVIEW

**Comment by bughit** [comment]
added docs for this and #12614

File: docs/api/web-frame.md [hunk 1]
```diff
@@ -235,6 +235,17 @@ Returns `WebFrame` - A child of `webFrame` with the supplied `name`, `null`
...
+### `webFrame.findFrameByRoutingId(routingId)`
+
+* `routingId` Integer - A unique frame id in the current renderer process.
+  Routing IDs can be retrieved from `WebFrame` instances (`webFrame.routingId`)
+  and are also passed by frame specific `WebContents` navigation events (e.g. 
+  `did-frame-navigate`)
+  
+Returns `WebFrame` - that has the supplied `routingId`, `null` if not found. 
+
+Returns
```
Comment (zcbenz) [inline docs/api/web-frame.md]: Redundant line.

**Comment by zcbenz** [comment]
And do you mind rebasing this on `master`? We had a new change that does formatting check and it may require modifications for this PR.

**Comment by bughit** [comment]
fixed and rebased

**Pull request merged**